src/utils/variant_helpers.hpp

```diff
@@ -0,0 +1,22 @@
+// Copyright 2021 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#pragma once
+
+namespace utils {
+template <class... Ts>
+struct Overloaded : Ts... {
+  using Ts::operator()...;
+};
+
+template <class... Ts>
+Overloaded(Ts...) -> Overloaded<Ts...>;
```

Comment:
You can drop the deduction guide if you add a consturctor that ignores its arguments `Overloaded(Ts...) {}`. Deduction guides are more explicit. However with a constructor you could have everything defined inside the class body (which I think is a little bit more readable? I don't have a strong preference, so please choose the one you think it's better. I making this comments for completeness :)

Reply:
I would leave it as is as it's from the `cppreference` :smile: The best case would be if we have nothing at all, so let's hope Clang fixes this.

Reply:
Sounds good :muscle: